src/_pytest/python_api.py

```diff
@@ -164,12 +164,13 @@ def get_value_from_nested_list(
         )
 
         # convert other_side to numpy array to ensure shape attribute is available
-        other_side = _as_numpy_array(other_side)
+        other_side_as_array = _as_numpy_array(other_side)
+        assert other_side_as_array is not None
```

Comment:
can we trigger this with a mistaken compare? if yes i think we need to add a super call if thats possible (unfortunately im not deeply familiar with the interaction of approx objects)

Reply:
what happens when a non-numpy array object is compared with the object in question

Reply:
I wrote a test case for it:
```
def test_numpy_array_with_non_numpy_array(self):
        np = pytest.importorskip("numpy")

        a1 = np.array([[1, 2]])
        a2 = [[[1, 2]]]
        assert a1 == a2
```
And it will fail:
```
___________________________________________________________ TestApprox.test_numpy_array_with_non_numpy_array ____________________________________________________________

self = <approx.TestApprox object at 0x10483ea40>

    def test_numpy_array_with_non_numpy_array(self):
        np = pytest.importorskip("numpy")
    
        a1 = np.array([[1, 2]])
        a2 = [1, 2]
>       assert a1 == a2
E       ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

testing/python/approx.py:771: ValueError
```

Reply:
excellent, thanks for the demonstration :)

Reply:
Would there be any additional changes I would have to make now?